### PR TITLE
Add tests for impl/auth.go and impl/context.go

### PR DESCRIPTION
**Pull request by TimothyGu**

### CODE REVIEW

File: server/Makefile [hunk 1]
```diff
@@ -12,5 +12,5 @@ format:
 
 .PHONY: generate
 generate:
-	go generate ./...
+	go generate -x ./...
```
Comment (TimothyGu) [inline server/Makefile:15]: This displays all commands executed during `go generate`.

File: server/impl/context.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ import (
 	"github.com/gorilla/sessions"
 )
 
+//go:generate go run golang.org/x/tools/cmd/stringer -type=contextKey
```
Comment (TimothyGu) [inline server/impl/context.go:9]: [stringer](https://pkg.go.dev/golang.org/x/tools/cmd/stringer) automatically generates a .String() method for contextKey based on the constant names.

File: server/impl/tools.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// +build tools
+
+package impl
+
+import (
+	_ "golang.org/x/tools/cmd/stringer"
```
Comment (TimothyGu) [inline server/impl/tools.go:6]: This declares a dependency on the stringer CLI tool, which is needed for the `go:generate` lines elsewhere.

File: server/impl/impl.go [hunk 4]
```diff
@@ -8,36 +8,53 @@ import (
...
 	sessionStoreInit sync.Once
 }
 
+// NewImplementation returns a new Implementation intended for production,
+// with a sessions.CookieStore as the internal session store.
 func NewImplementation() *Implementation {
 	return new(Implementation)
 }
 
+// NewImplementation returns a new Implementation with the provided session store.
```
Comment (jamieliu386) [inline server/impl/impl.go:27]: I assume this should be NewImplementationWithSessionStore
```suggestion
// NewImplementationWithSessionStore returns a new Implementation with the provided session store.
```

**Review by jamieliu386 — APPROVED** [review]
This is really helpful, thanks!

File: server/impl/impl.go [hunk 5]
```diff
@@ -8,36 +8,53 @@ import (
...
 	}
 
-	sessionStore     *sessions.CookieStore // lazily initialized; use SessionStore() instead!
+	sessionStore     sessions.Store // could be lazily initialized; use SessionStore() instead!
 	sessionStoreInit sync.Once
 }
 
+// NewImplementation returns a new Implementation intended for production,
+// with a sessions.CookieStore as the internal session store.
 func NewImplementation() *Implementation {
```
Comment (Timthetic) [inline server/impl/impl.go:23]: I don't see this being used anywhere. Right now, I'm assuming that this is here to satisfy some interface.

**Review by Timthetic — APPROVED** [review]

**Review by Timthetic — APPROVED** [review]
LGTM

File: server/impl/impl.go (hunk 5 above)
Comment (TimothyGu) [reply server/impl/impl.go:23]: It's used for actual server startup; see https://github.com/downToMeet/downToMeet/blob/d2c28f1cf84a3e295ec2ff8cd05ef8f019bf7906/server/restapi/configure_down_to_meet.go#L21.

**Pull request merged**